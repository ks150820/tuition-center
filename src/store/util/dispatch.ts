import {apiCallBegan} from '../actions/actions';
import {store} from '../configureStore';
interface IDispatchType {
  url: string;
  method: string;
  onStart: string;
  onSuccess: string;
  onError: string;
  lastCalledTime: number;
  cacheValidityDuration: number;
}

const storeDispatch = (payload: IDispatchType) => {
  return store.dispatch({type: apiCallBegan.type, payload: payload});
};
export default storeDispatch;
