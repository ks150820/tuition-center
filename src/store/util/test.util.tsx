import React from 'react';
import {
  render as rtlRender,
  RenderAPI,
  renderHook as rtlRenderHook,
} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import reducers from '../reducer';

const render = (
  component: React.ReactElement<any>,
  {
    preloadedState,
    store = configureStore({reducer: reducers, preloadedState}),
    ...renderOptions
  }: any = {},
): RenderAPI => {
  const wrapper = ({children}: any) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return rtlRender(component, {wrapper: wrapper, ...renderOptions});
};

const renderHook = (
  viewController: any,
  args: any = {},
  preloadedState?: any,
  ProviderWrapper?: any,
  providerWrapperArgs?: any,
) => {
  const wrapper = ({children}: any) => {
    const store = configureStore({
      reducer: reducers,
      preloadedState,
      middleware: (getDefaultMiddleware: any) => [
        ...getDefaultMiddleware({serializableCheck: false}),
      ],
    });

    if (ProviderWrapper) {
      return (
        <Provider store={store}>
          <ProviderWrapper {...providerWrapperArgs}>{children}</ProviderWrapper>
        </Provider>
      );
    }

    return <Provider store={store}>{children}</Provider>;
  };

  const {result} = rtlRenderHook(() => viewController(args), {wrapper});

  return {
    result,
  };
};

export {render, renderHook};
