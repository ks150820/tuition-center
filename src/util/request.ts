import {requestMethodType} from '../store/@types/api/api';

type progressHandlerType = (progressEvent: number) => {};
interface IRequestType {
  url: string; // `url` is the server URL that will be used for the request
  method: requestMethodType; // `method` is the request method to be used when making the request
  // `baseURL` will be prepended to `url` unless `url` is absolute.
  // It can be convenient to set `baseURL` for an instance of axios to pass relative URLs
  // to methods of that instance.
  baseURL: string;
  data?: {};
  // `timeout` specifies the number of milliseconds before the request times out.
  // If the request takes longer than `timeout`, the request will be aborted.
  timeout?: 0 | 1000 | number; // default is `0` (no timeout)
  // `onUploadProgress` allows handling of progress events for uploads
  // browser only
  onUploadProgress?: progressHandlerType;

  // `onDownloadProgress` allows handling of progress events for downloads
  // browser only
  onDownloadProgress?: progressHandlerType;
}

export const createRequestObject = (
  url: string,
  method: requestMethodType,
  data?: object,
  timeout?: number,
  onUploadProgressHandler?: progressHandlerType,
  onDownloadProgressHandler?: progressHandlerType,
) => {
  const request: IRequestType = {
    url: url,

    method: method, // default

    baseURL: 'https://mocki.io/v1/',
  };

  if (data) {
    request.data = data;
  }
  if (timeout) {
    request.timeout = timeout;
  }
  if (onUploadProgressHandler) {
    request.onUploadProgress = onUploadProgressHandler;
  }
  if (onDownloadProgressHandler) {
    request.onDownloadProgress = onDownloadProgressHandler;
  }

  return request;
};
