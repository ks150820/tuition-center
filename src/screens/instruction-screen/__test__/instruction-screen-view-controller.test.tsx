import {waitFor} from '@testing-library/react-native';
import {renderHook} from '@store/util/test.util';
import useInstructionScreenController from '@screens/instruction-screen/instruction-screen-view-controller';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

describe('instruction-screen-view-controller', () => {
  test('test component for English option value and test for loader and languages', () => {
    const {result} = renderHook(
      useInstructionScreenController,
      {},
      {
        entities: {
          testExperience: {
            loading: true,
          },
        },
      },
    );

    waitFor(() => {
      result.current.handleTestShowOption();
      result.current.handleShowOptions();
      result.current.handleTermsData(true);
      result.current.handleOptionValue({name: 'English', id: 'en'});
      result.current.handleTestLanguage({name: 'English', id: 'en'});
      result.current.handleStartTest();
      result.current.handleTerms('test');
      expect(result.current.loader).toBe(true);
      expect(result.current.languages).toMatchSnapshot([
        {name: 'English', id: 'en'},
        {name: 'Hindi', id: 'hi'},
      ]);
    });
  });
  test('test component for hindi option', () => {
    const {result} = renderHook(
      useInstructionScreenController,
      {},
      {
        entities: {
          testExperience: {
            loading: true,
          },
        },
      },
    );

    waitFor(() => {
      result.current.handleOptionValue({name: 'Hindi', id: 'hi'});
    });
  });
});
