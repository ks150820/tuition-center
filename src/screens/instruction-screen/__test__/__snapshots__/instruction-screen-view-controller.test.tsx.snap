// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruction-screen-view-controller test component for English option value and test for loader and languages 1`] = `
Array [
  Object {
    "id": "en",
    "name": "English",
  },
  Object {
    "id": "hi",
    "name": "Hindi",
  },
]
`;
